tuf_conformance/simulator_server.py

```diff
@@ -54,8 +54,17 @@ def __init__(self):
         # key is test name, value is the repository sim for that test
         self.repos: Dict[str, RepositorySimulator] = {}
 
-    def get_client_init_data(self, repo: str) -> ClientInitData:
-        return ClientInitData(
-            f"http://{self.server_address[0]}:{self.server_address[1]}/{repo}/metadata/",
-            self.repos[repo].fetch_metadata("root", 1)
+    def new_test(self, name: str) -> tuple[ClientInitData, RepositorySimulator]:
+        """Return a tuple of
+        * A new repository simulator (for test case to control)
+        * client initialization parameters (so client can find the simulated repo)
+        """
+        repo = RepositorySimulator()
+        self.repos[name] = repo
+
+        client_data = ClientInitData(
+            f"http://{self.server_address[0]}:{self.server_address[1]}/{name}/metadata/",
```

Comment:
should this be `parse.quote(name)`? Ruby's `URI` complains about unescaped brackets `[]` in the URI

Reply:
you are right and then it needs to be unquoted when the request is handled (I'll take care of that in #98)